ui/app/src/modules/Content/Delete/DeleteDialog.tsx

```diff
@@ -272,18 +279,24 @@ function DeleteDialogWrapper(props: DeleteDialogProps) {
   });
 
   useEffect(() => {
-    fetchDeleteDependencies(siteId, selectedItems).subscribe(
-      (response: any) => {
-        setDeleteDependencies({
-          childItems: response.items.childItems,
-          dependentItems: response.items.dependentItems
-        });
-      },
-      (error) => {
-        setApiState({ error });
+    if (selectedItems.length) {
+      fetchDeleteDependencies(siteId, selectedItems).subscribe(
+        (items) => {
+          setDeleteDependencies({
+            childItems: items.childItems,
+            dependentItems: items.dependentItems
+          });
+        },
+        (error) => {
+          setApiState({ error });
+        }
+      );
+
+      if (fetch) {
+        dispatch(fetchedDepsDeleteDialog());
```

Comment:
Let's talk through this.
 - Location / effect mix up
 - Purpose
 - Action name